### DISCUSSION TITLE
Does delta-rs support safe writing to azure file share?

### DISCUSSION THREAD

**Discussion by PeterThramkrongart** [delta-io/delta-rs#3054] [Q&A]
I'm trying to write Delta tables to an Azure File Share mount point, but encountering rename operation errors both locally and in Kubernetes. Adding `storage_options={"allow_unsafe_rename":"true"}` allows writes to work, but this feels like a workaround.

Is Azure File Share officially supported as a storage backend for Delta tables?

### Error in Kubernetes:
OSError: Generic LocalFileSystem error: Unable to rename file: Operation not supported (os error 95)

### Error on Windows with mounted drive:
OSError: Generic LocalFileSystem error: Unable to rename file: The parameter is incorrect. (os error 87)

### Environment:
- Python 3.11
- deltalake==0.21.0
- Azure File Share (mounted in both Windows and Debian/Kubernetes)

**Comment by rtyler** [comment]
I assume that Azure File Share is CIFS under the covers? I'm curious what URL scheme is being used here, that would give me a hint as to what underlying storage provider might be used.

We do support Azure blob storage, but I'm unfamiliar with the differences between Azure File Share and other things in the Azure cloud offerings
